lib/readline.js

```diff
@@ -868,6 +864,14 @@ Interface.prototype._ttyWrite = function(s, key) {
         break;
 
       default:
+        if (key.name === 'tab') {
+          // If tab completion enabled, do that...
+          if (typeof this.completer === 'function') {
+            this._tabComplete();
+            break;
+          }
+        }
```

Comment:
Perhaps a good solution is to extract the `default` block contents to its own function. The `default` block will run that function, of course, but so will the `tab` block if `completer` is undefined? So instead of falling through to `default`, the `tab` block would `break` like every other block, satisfying the linter.

On the other hand, the `default` block logic is not that complicated (yet, at least), and it is nice when reading a looooonnng `switch/case` like that to not have to go hunting for the default function if it's short. So I could go either way on this.